CI required jobs are cancelled after one-hour cli-core-rest/coverage hangs

## Summary

Internal CI has repeatedly cancelled the same required jobs after roughly one hour on otherwise unrelated PRs, leaving mergeable PRs blocked by `CI Status` without a code-specific failure.

Observed on:

- #1880 head `8adc8c6e`, CI run `24874397549`
- #1863 head `987b7117`, CI run `24874398474`

Both runs show the same pattern:

- `Test (Node 20 / cli-core-rest)` starts and remains in progress until cancelled
- `Coverage Report (TypeScript Full)` starts and is cancelled
- `CI Status` fails because those required jobs are cancelled
- other required jobs pass (`rustfmt`, `clippy`, `lint`, `typecheck`, Rust coverage, team critical coverage, Ralph persistence, smoke, etc.)

This also happened earlier on #1863 before a no-source retrigger commit, so this is not isolated to a single PR head.

## Evidence

#1880 run `24874397549`:

```text
CI Status: test: cancelled
CI Status: coverage-ts-full: cancelled
CI Status: build: skipped
One or more required CI jobs failed
```

#1863 run `24874398474`:

```text
CI Status: test: cancelled
CI Status: coverage-ts-full: cancelled
CI Status: build: skipped
One or more required CI jobs failed
```

The coverage job logs reach normal passing test output and then end with:

```text
##[error]The operation was canceled.
Cleaning up orphan processes
Terminate orphan process: npm run coverage:ts:full:compiled
```

The `cli-core-rest` cancelled job logs are not available through `gh run view --job ... --log` after cancellation, which makes the hang opaque.

## Expected

The required CI lanes should either complete or fail with actionable logs. Mergeable PRs should not be blocked by a repeated one-hour cancellation pattern in shared CI lanes.

## Suggested investigation

- Inspect `cli-core-rest` for a hanging test/process leak.
- Inspect `coverage:ts:full:compiled` for worker/process leaks around the tail of the suite.
- Ensure cancelled required jobs leave enough logs/artifacts to diagnose the hung command.
- Consider splitting or timeout-wrapping the long lanes so the failing test/file is visible.

## Current blocked PRs

- #1880 (`Fixes #1879`)
- #1863

—
*[repo owner's gaebal-gajae (clawdbot) 🦞]*

--- a/src/cli/__tests__/setup-agents-overwrite.test.ts
+++ b/src/cli/__tests__/setup-agents-overwrite.test.ts
@@ -49,6 +49,7 @@ async function runSetupWithCapturedLogs(
   };
   try {
     await setup({
+      installModePrompt: async (defaultMode) => defaultMode,
       modelUpgradePrompt: async () => false,
       ...options,
     });

--- a/src/scripts/__tests__/run-test-files.test.ts
+++ b/src/scripts/__tests__/run-test-files.test.ts
@@ -5,16 +5,15 @@ import { join } from 'node:path';
 import { describe, it } from 'node:test';
 import assert from 'node:assert/strict';
 
-function runCompiledRunner(root: string, timeoutMs: number) {
+function runCompiledRunner(root: string, envOverrides: Record<string, string> = {}, timeoutMs = 5_000) {
   return spawnSync(process.execPath, ['dist/scripts/run-test-files.js', root], {
     cwd: process.cwd(),
     encoding: 'utf-8',
     env: {
       ...process.env,
-      OMX_NODE_TEST_TIMEOUT_MS: String(timeoutMs),
-      OMX_NODE_TEST_RUNNER_TIMEOUT_MS: String(timeoutMs + 500),
+      ...envOverrides,
     },
-    timeout: timeoutMs + 5_000,
+    timeout: timeoutMs,
   });
 }
 
@@ -34,12 +33,38 @@ describe('run-test-files diagnostics', () => {
         ].join('\n'),
       );
 
-      const result = runCompiledRunner(wd, 250);
+      const result = runCompiledRunner(wd, {
+        OMX_NODE_TEST_TIMEOUT_MS: '250',
+        OMX_NODE_TEST_RUNNER_TIMEOUT_MS: '750',
+      });
 
       assert.notEqual(result.status, 0);
       assert.match(result.stderr, /per-test timeout 250ms/);
       assert.match(result.stderr, /node --test did not exit normally|runner timeout 750ms/);
       assert.match(`${result.stdout}\n${result.stderr}`, /hang\.test\.js|never resolves|cancelled/i);
+    } finally {
+      rmSync(wd, { recursive: true, force: true });
+    }
+  });
+
+  it('logs that per-test timeout is disabled by default', () => {
+    const wd = mkdtempSync(join(tmpdir(), 'omx-run-test-files-'));
+    try {
+      const testsDir = join(wd, '__tests__');
+      mkdirSync(testsDir, { recursive: true });
+      writeFileSync(
+        join(testsDir, 'pass.test.js'),
+        [
+          "import { test } from 'node:test';",
+          "test('passes', () => {});",
+          '',
+        ].join('\n'),
+      );
+
+      const result = runCompiledRunner(wd);
+
+      assert.equal(result.status, 0, result.stderr || result.stdout);
+      assert.match(result.stderr, /per-test timeout disabled/);
     } finally {
       rmSync(wd, { recursive: true, force: true });
     }

--- a/src/scripts/__tests__/run-test-files.test.ts
+++ b/src/scripts/__tests__/run-test-files.test.ts
@@ -1,0 +1,47 @@
+import { spawnSync } from 'node:child_process';
+import { mkdirSync, mkdtempSync, writeFileSync, rmSync } from 'node:fs';
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+import { describe, it } from 'node:test';
+import assert from 'node:assert/strict';
+
+function runCompiledRunner(root: string, timeoutMs: number) {
+  return spawnSync(process.execPath, ['dist/scripts/run-test-files.js', root], {
+    cwd: process.cwd(),
+    encoding: 'utf-8',
+    env: {
+      ...process.env,
+      OMX_NODE_TEST_TIMEOUT_MS: String(timeoutMs),
+      OMX_NODE_TEST_RUNNER_TIMEOUT_MS: String(timeoutMs + 500),
+    },
+    timeout: timeoutMs + 5_000,
+  });
+}
+
+describe('run-test-files diagnostics', () => {
+  it('applies a bounded node --test timeout so hanging tests fail with file context', () => {
+    const wd = mkdtempSync(join(tmpdir(), 'omx-run-test-files-'));
+    try {
+      const testsDir = join(wd, '__tests__');
+      mkdirSync(testsDir, { recursive: true });
+      const testPath = join(testsDir, 'hang.test.js');
+      writeFileSync(
+        testPath,
+        [
+          "import { test } from 'node:test';",
+          "test('never resolves', async () => { await new Promise(() => setInterval(() => {}, 1_000)); });",
+          '',
+        ].join('\n'),
+      );
+
+      const result = runCompiledRunner(wd, 250);
+
+      assert.notEqual(result.status, 0);
+      assert.match(result.stderr, /per-test timeout 250ms/);
+      assert.match(result.stderr, /node --test did not exit normally|runner timeout 750ms/);
+      assert.match(`${result.stdout}\n${result.stderr}`, /hang\.test\.js|never resolves|cancelled/i);
+    } finally {
+      rmSync(wd, { recursive: true, force: true });
+    }
+  });
+});

--- a/src/scripts/run-test-files.ts
+++ b/src/scripts/run-test-files.ts
@@ -2,7 +2,7 @@ import { spawnSync } from 'node:child_process';
 import { readdirSync, statSync } from 'node:fs';
 import { join, resolve } from 'node:path';
 
-const DEFAULT_TEST_TIMEOUT_MS = 120_000;
+const DEFAULT_TEST_TIMEOUT_MS = 0;
 const DEFAULT_RUNNER_TIMEOUT_MS = 30 * 60 * 1_000;
 
 function collectTests(path: string, out: string[]): void {

--- a/src/scripts/run-test-files.ts
+++ b/src/scripts/run-test-files.ts
@@ -2,6 +2,9 @@ import { spawnSync } from 'node:child_process';
 import { readdirSync, statSync } from 'node:fs';
 import { join, resolve } from 'node:path';
 
+const DEFAULT_TEST_TIMEOUT_MS = 120_000;
+const DEFAULT_RUNNER_TIMEOUT_MS = 30 * 60 * 1_000;
+
 function collectTests(path: string, out: string[]): void {
   let stats;
   try {
@@ -22,6 +25,13 @@ function collectTests(path: string, out: string[]): void {
   }
 }
 
+function parseTimeoutMs(value: string | undefined, defaultTimeoutMs: number): number {
+  if (!value) return defaultTimeoutMs;
+  const parsed = Number(value);
+  if (!Number.isFinite(parsed) || parsed < 0) return defaultTimeoutMs;
+  return Math.floor(parsed);
+}
+
 const roots = process.argv.slice(2);
 const targets = roots.length > 0 ? roots : ['dist'];
 const files: string[] = [];
@@ -36,13 +46,41 @@ if (files.length === 0) {
   process.exit(1);
 }
 
-const result = spawnSync(process.execPath, ['--test', ...files], {
+const testTimeoutMs = parseTimeoutMs(process.env.OMX_NODE_TEST_TIMEOUT_MS, DEFAULT_TEST_TIMEOUT_MS);
+const runnerTimeoutMs = parseTimeoutMs(process.env.OMX_NODE_TEST_RUNNER_TIMEOUT_MS, DEFAULT_RUNNER_TIMEOUT_MS);
+const testArgs = ['--test'];
+if (testTimeoutMs > 0) {
+  testArgs.push(`--test-timeout=${testTimeoutMs}`);
+}
+testArgs.push(...files);
+
+console.error(
+  `[run-test-files] running ${files.length} test file(s) from ${targets.join(', ')}${
+    testTimeoutMs > 0 ? ` with per-test timeout ${testTimeoutMs}ms` : ' with per-test timeout disabled'
+  }${runnerTimeoutMs > 0 ? ` and runner timeout ${runnerTimeoutMs}ms` : ' and runner timeout disabled'}`,
+);
+
+const childEnv = { ...process.env };
+delete childEnv.NODE_TEST_CONTEXT;
+
+const result = spawnSync(process.execPath, testArgs, {
   stdio: 'inherit',
-  env: process.env,
+  env: childEnv,
+  timeout: runnerTimeoutMs > 0 ? runnerTimeoutMs : undefined,
+  killSignal: 'SIGTERM',
 });
 
 if (typeof result.status === 'number') {
   process.exit(result.status);
 }
 
+if (result.error) {
+  console.error(`[run-test-files] node --test error: ${result.error.message}`);
+}
+console.error(
+  `[run-test-files] node --test did not exit normally${result.signal ? ` (signal: ${result.signal})` : ''}. `
+    + `Roots: ${targets.join(', ')}. Test files: ${files.length}. `
+    + `Per-test timeout: ${testTimeoutMs > 0 ? `${testTimeoutMs}ms` : 'disabled'}. `
+    + `Runner timeout: ${runnerTimeoutMs > 0 ? `${runnerTimeoutMs}ms` : 'disabled'}.`,
+);
 process.exit(1);

